File: python/RandomAgent.py
from Gomoku import *
import random

class RandomAgent:
    def __init__(self):
        pass
    
    def play(self, board_state):
        row = random.randrange(0, BOARD_SIZE)
        col = random.randrange(0, BOARD_SIZE)
        while abs(board_state[row][col]) > 1e-3:
            row = random.randrange(0, BOARD_SIZE)
            col = random.randrange(0, BOARD_SIZE)
       
        board_state.turn += 1
        board_state[row][col] = board_state.turn
        board_state.last_row = row
        board_state.last_col = col

        return board_state
       

File: cpp/random_agent.cpp
#include "random_agent.h"

#include <stdlib.h>

BoardState RandomAgent::play(BoardState board_state)
{
	int row = rand() % BOARD_SIZE;
	int col = rand() % BOARD_SIZE;

	while(board_state[row][col] != 0) {
		row = rand() % BOARD_SIZE;
		col = rand() % BOARD_SIZE;
	}

	board_state.turn++;
	board_state[row][col] = board_state.turn;
	board_state.last_row = row;
	board_state.last_col = col;

	return board_state;
}
